classes/Address.php

```diff
@@ -303,7 +303,11 @@ public static function getZoneById($id_address)
 			LEFT JOIN `' . _DB_PREFIX_ . 'state` s ON s.`id_state` = a.`id_state`
 			WHERE a.`id_address` = ' . (int) $id_address);
 
-        self::$_idZones[$id_address] = (int) ((int) $result['id_zone_state'] ? $result['id_zone_state'] : $result['id_zone']);
+        if (empty($result['id_zone_state']) && empty($result['id_zone'])) {
+            return false;
+        }
+
+        self::$_idZones[$id_address] = (int) ((int) $result['id_zone_state'] ?? $result['id_zone']);
```

Comment:
right side of ?? never reached

```suggestion
        self::$_idZones[$id_address] = (int) ($result['id_zone_state'] ?? $result['id_zone']);
```

Reply:
Thanks, sometimes I forget they didn't learn about null value in the database and stored "0" :facepalm:

Reply:
may be true, but did you tested the removed int cast? It comes from join...